src/frontend/src/binder/expr/function/mod.rs

```diff
@@ -104,229 +119,246 @@ impl Binder {
         // retrieve object comment, however we don't support casting a non-literal expression to
         // regclass. We just hack the `obj_description` and `col_description` here, to disable it to
         // bind its arguments.
-        if function_name == "obj_description" || function_name == "col_description" {
+        if func_name == "obj_description" || func_name == "col_description" {
             return Ok(ExprImpl::literal_varchar("".to_string()));
         }
-        if function_name == "array_transform" {
+
+        // special binding logic for `array_transform`
+        if func_name == "array_transform" {
             // For type inference, we need to bind the array type first.
-            return self.bind_array_transform(f);
+            reject_syntax!(
```

Comment:
Seems heavy to me to introduce a macro instead of just a function to construct 
```
            return Err(ErrorCode::InvalidInputSyntax($msg.to_string()).into());
```

But I'm not going to block this PR with it.